src/coreclr/jit/lower.cpp

```diff
@@ -7976,6 +7976,70 @@ void Lowering::TransformUnusedIndirection(GenTreeIndir* ind, Compiler* comp, Bas
     }
 }
 
+//------------------------------------------------------------------------
+// LowerLclHeap: a common logic to lower LCLHEAP.
+//
+// Arguments:
+//    blkNode - the LCLHEAP node we are lowering.
+//
+void Lowering::LowerLclHeap(GenTree* node)
+{
+    assert(node->OperIs(GT_LCLHEAP));
+
+#if defined(TARGET_XARCH)
+    if (node->gtGetOp1()->IsCnsIntOrI())
+    {
+        GenTreeIntCon* sizeNode = node->gtGetOp1()->AsIntCon();
+        assert((unsigned)sizeNode->IconValue() <= UINT_MAX);
+        sizeNode->SetIconValue(ALIGN_UP(sizeNode->IconValue(), STACK_ALIGN));
+        if (comp->info.compInitMem)
+        {
+            ssize_t size = node->gtGetOp1()->AsIntCon()->IconValue();
+            if ((unsigned)size > UINT_MAX)
+            {
+                BADCODE("Localloc size is negative or too big");
+            }
+
+            LIR::Use use;
+            if (BlockRange().TryGetUse(node, &use))
+            {
+                // Replace with null for LCLHEAP(0)
+                if (size == 0)
+                {
+                    GenTree* nullNode = comp->gtNewIconNode(0, node->TypeGet());
+                    use.ReplaceWith(nullNode);
+                    BlockRange().Remove(sizeNode);
+                    BlockRange().Remove(node);
+                    BlockRange().InsertAfter(node, nullNode);
+                    return;
```

Comment:
```suggestion
                    node->BashToZeroConst(TYP_I_IMPL);
                    BlockRange().Remove(sizeNode);
                    return;
```

Reply:
I keep forgetting about this handfull BashTo* apis 🙂